integration-tests/cli/product-json-to-xlsx.it.js

```diff
@@ -0,0 +1,885 @@
+  // remove first column containing null values
+  rows.forEach(row => row.shift())
+
+  return {
+    workbook,
+    worksheet,
+    rows,
+  }
+}
+
+async function analyzeExcelFile(path) {
+  const workbook = new Excel.Workbook()
+  await workbook.xlsx.readFile(path)
+  return {
+    path,
+    ... await analyzeExcelWorkbook(workbook),
+  }
+}
+
+describe('XLSX and CLI Tests', () => {
+  let apiConfig
+  const binPath = './integration-tests/node_modules/.bin/product-json-to-xlsx'
+
+  beforeAll(async () => {
+    const credentials = await getCredentials(projectKey)
+    apiConfig = {
+      host: 'https://auth.sphere.io',
+      apiUrl: 'https://api.sphere.io',
+      projectKey,
+      credentials,
+    }
+
+    await cleanup(apiConfig)
+
+    // Create data on API
+    await createData(apiConfig, 'productTypes', [
+      sampleProductType,
+      anotherSampleProductType,
+    ])
+
+    await createData(apiConfig, 'categories', [sampleParentCategory])
+    await createData(apiConfig, 'categories', [sampleCategory])
+
+    const priceChannel = await createData(apiConfig, 'channels', [
+      samplePriceChannel,
+    ])
+    const state = await createData(apiConfig, 'states', [sampleState])
+    const taxCategory = await createData(apiConfig, 'taxCategories', [
+      sampleTaxCategory,
+    ])
+
+    const stateRef = {
+      typeId: 'state',
+      id: state[0].body.id,
+    }
+    const taxCategoryRef = {
+      typeId: 'tax-category',
+      id: taxCategory[0].body.id,
+    }
+
+    const sampleProducts = createProducts(stateRef, taxCategoryRef)
+    sampleProducts[0].masterVariant.prices[0].channel = {
+      typeId: 'channel',
+      id: priceChannel[0].body.id,
+    }
+
+    await createData(apiConfig, 'products', sampleProducts)
+  }, 20000)
+
+  afterAll(() => cleanup(apiConfig))
+
+  describe('CLI basic functionality', () => {
+    it('should print usage information given the help flag', async () => {
+      const [stdout, stderr] = await exec(`${binPath} --help`)
+      expect(stdout).toMatchSnapshot()
+      expect(stderr).toBeFalsy()
+    })
+
+    it('should print the module version given the version flag', async () => {
+      const [stdout, stderr] = await exec(`${binPath} --version`)
+      expect(stdout).toBe(`${version}\n`)
+      expect(stderr).toBeFalsy()
+    })
+  })
+
+  describe('Parser', () => {
+    const exporter = './integration-tests/node_modules/.bin/product-exporter'
+    describe('From product exporter', () => {
+      describe('WITHOUT HEADERS:: write products to `zip` file', () => {
+        let product1
+        let product2
+        const fileNames = []
+
+        beforeAll(async done => {
+          const zipFile = tmp.fileSync({ postfix: '.zip' }).name
+          const [stdout, stderr] = await exec(
+            `${exporter} -p ${projectKey} -s | ${binPath} -p ${projectKey} --referenceCategoryBy namedPath --fillAllRows -o ${zipFile}`
+          )
+          expect(stdout).toBeTruthy()
```

Comment:
hmm assertion inside the `beforeAll`? Why not inside a test?

Reply:
This came from product-json-to-csv .. I removed it here 233983c from xlsx and here 27c2288 from csv mapper

Reply:
This came from product-json-to-csv .. I removed it here 233983c from xlsx and here 27c2288 from csv mapper